config-cmd/help.go

```diff
@@ -0,0 +1,98 @@
+package configCmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/spf13/cobra"
+	"github.com/taskcluster/taskcluster-cli/config"
+)
+
+func init() {
+	Command.AddCommand(&cobra.Command{
+		Use:   "help [<option>]",
+		Short: "Get help for TaskCluster's configuration options.",
+		RunE:  cmdHelp,
+	})
+}
+
+func cmdHelp(cmd *cobra.Command, args []string) error {
+	// default, print help for all commands
+	if len(args) == 0 {
+		printHelp(cmd)
+		return nil
+	}
+
+	// otherwise, print for one specific option
+	err := printOptionHelp(cmd, args[0])
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// printOptionHelp shows help for specific option
+func printOptionHelp(cmd *cobra.Command, key string) error {
+	command, option, definition, value, err := getOptionFromKey(key)
+	if err != nil {
+		return err
+	}
+
+	defaultValue := definition.Default
+	if definition.Parse {
+		if s, err := json.MarshalIndent(defaultValue, "  ", "  "); err == nil {
+			defaultValue = "\n" + string(s)
+		} else {
+			defaultValue = fmt.Sprintf("%#v", defaultValue)
+		}
+		if s, err := json.MarshalIndent(value, "  ", "  "); err == nil {
+			value = "\n" + string(s)
+		} else {
+			value = fmt.Sprintf("%#v", value)
+		}
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Key:     %s.%s\n", command, option)
+	fmt.Fprintln(cmd.OutOrStdout(), definition.Description)
+	fmt.Fprintf(cmd.OutOrStdout(), "Value:   %s\n", value)
+	fmt.Fprintf(cmd.OutOrStdout(), "Default: %s\n", defaultValue)
+
+	return nil
+}
+
+// printHelp as a list of all possible configuration options
+func printHelp(cmd *cobra.Command) {
+	commands := []string{}
+	maxLength := 0
+
+	for command, options := range config.OptionsDefinitions {
+		// this is for the list of commands
+		commands = append(commands, command)
+
+		// this is for the maximum name length for a key
+		for option, _ := range options {
```

Comment:
nit: should omit 2nd value from range; this loop is equivalent to `for option := range ...` (golint)